From: Keith Busch <keith.busch@intel.com>
Date: Fri, 17 Jul 2015 14:43:03 -0600
Subject: [PATCH] NVMe: Fix ioctl passthrough length calculation
Patch-mainline: Never, in nvme-legacy repository
Git-commit: 3882a5f2e6335fca52c1389d5eb5eccc79cc1562
Git-repo: http://git.infradead.org/users/kbusch/nvme-legacy.git
References: bsc#948374

The length was previously verified when the prps were setup. Checking
it again against the lba size breaks when using extended meta-data.

Signed-off-by: Keith Busch <keith.busch@intel.com>
Acked-by: Johannes Thumshirn <jthumshirn@suse.de>
---
 drivers/block/nvme-core.c |    5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

--- a/drivers/block/nvme-core.c
+++ b/drivers/block/nvme-core.c
@@ -1766,10 +1766,7 @@ static int nvme_submit_io(struct nvme_ns
 	c.rw.prp1 = cpu_to_le64(sg_dma_address(iod->sg));
 	c.rw.prp2 = cpu_to_le64(iod->first_dma);
 
-	if (length != (io.nblocks + 1) << ns->lba_shift)
-		status = -ENOMEM;
-	else
-		status = nvme_submit_io_cmd(dev, &c, NULL);
+	status = nvme_submit_io_cmd(dev, &c, NULL);
 
 	if (meta_len) {
 		if (status == NVME_SC_SUCCESS && !(io.opcode & 1)) {
